File: application/src/test/java/com/github/francescojo/ApplicationTest.java
/*
 * boot-gs-multi-module skeleton.
 * Under no licences and warranty.
 */
package com.github.francescojo;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.ApplicationContext;
import org.springframework.test.context.junit4.SpringRunner;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Instrumentation test definition for boot-gs-multi-module
 *
 * @author Francesco Jo(nimbusob@gmail.com)
 * @since 01 - Jan - 2018
 */
@RunWith(SpringRunner.class)
@SpringBootTest
public class ApplicationTest {
    @Autowired
    private ApplicationContext myAppContext;

    @Test
    public void contextLoads() {
        assertThat(myAppContext).isNotNull();
    }
}

File: app_main/src/test/kotlin/com/github/fj/restapi/ApplicationTest.kt
/*
 * springboot-multimodule-kotlin skeleton.
 * Under no licences and warranty.
 */
package com.github.fj.restapi

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.ApplicationContext
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * @author Francesco Jo(nimbusob@gmail.com)
 * @since 25 - Oct - 2018
 */
@ExtendWith(SpringExtension::class)
@SpringBootTest
class ApplicationTest {
    @Autowired
    private lateinit var context: ApplicationContext

    @Test
    fun `Spring context loads successfully`() {
        assertNotNull(context)
    }
}
